File: src/main/java/com/dongfg/project/api/graphql/handler/ExceptionWhileDataFetchingWrapper.java
package com.dongfg.project.api.graphql.handler;

import com.fasterxml.jackson.annotation.JsonIgnore;
import graphql.ErrorType;
import graphql.ExceptionWhileDataFetching;
import graphql.execution.ExecutionPath;
import graphql.language.SourceLocation;

import java.util.Map;

@SuppressWarnings("EmptyMethod")
class ExceptionWhileDataFetchingWrapper extends ExceptionWhileDataFetching {
    ExceptionWhileDataFetchingWrapper(ExecutionPath path, Throwable exception, SourceLocation sourceLocation) {
        super(path, exception, sourceLocation);
    }

    @JsonIgnore
    @Override
    public Throwable getException() {
        return super.getException();
    }

    @JsonIgnore
    @Override
    public Map<String, Object> getExtensions() {
        return super.getExtensions();
    }

    @JsonIgnore
    @Override
    public ErrorType getErrorType() {
        return super.getErrorType();
    }
}

File: src/main/kotlin/com/dongfg/project/api/web/graphql/handler/ExceptionWhileDataFetchingWrapper.kt
package com.dongfg.project.api.web.graphql.handler

import com.fasterxml.jackson.annotation.JsonIgnore
import graphql.ErrorType
import graphql.ExceptionWhileDataFetching
import graphql.execution.ExecutionPath
import graphql.language.SourceLocation

/**
 * @author dongfg
 * @date 2018/3/23
 */
class ExceptionWhileDataFetchingWrapper(path: ExecutionPath?, exception: Throwable?, sourceLocation: SourceLocation?)
    : ExceptionWhileDataFetching(path, exception, sourceLocation) {

    @JsonIgnore
    override fun getErrorType(): ErrorType {
        return super.getErrorType()
    }

    @JsonIgnore
    override fun getExtensions(): MutableMap<String, Any> {
        return super.getExtensions()
    }

    @JsonIgnore
    override fun getException(): Throwable {
        return super.getException()
    }
}